docs: break apart the ambient getting start guide (WIP) (#15168)
istio/istio.io: content/en/docs/ambient/getting-started/setup/index.md

---

> title: Install Ambient

Reviewer: ```suggestion
title: Install Istio with support for ambient mode
```

> description: Download and Install Istio in Ambient mode.

Reviewer: ```suggestion
description: Download and install Istio with support for ambient mode.
```

weight: 1
owner: istio/wg-networking-maintainers
test: yes
---

> In this section you'll download Istio CLI (`istioctl`) and install Istio in ambient mode on your Kubernetes cluster. You'll need a Kubernetes cluster to proceed. If you don't have a cluster, you can use [kind](/docs/setup/platform-setup/kind) or any other [supported Kubernetes platform](/docs/setup/platform-setup).

Reviewer: terminology note: "with support for ambient mode"

Reviewer: (ensure the kind docs are revised as part of this PR)

## 1. Download Istio CLI

> You'll use [Homebrew](https://brew.sh/) to download the Istio CLI:

Reviewer 2: ```suggestion
You'll use [Homebrew](https://brew.sh/) to download the Istio CLI:
```

Personally don't like doing this, we should be encouraging users to download the binary from us with a version that matches what their cluster is running. We should link to this as a pre-req, or mirror the same instructions for consistency. https://istio.io/latest/docs/setup/getting-started/#download

Author: One of the things I tried to achieve with these instructions is to simplify them and get the reader ready to go as fast as possible. Executing a download script that downloads a compressed file, unpacks it into the current folder then asks you to set up your dotfiles/path to make it useful doesn't help with simplicity. The fact that we're not using either Brew or have a simpler installation might be a separate thing that could be improved. 

Note that brew automatically and on a schedule checks for outdated packages and updates them. (e.g. the 1.22 was out on May 13th and the brew formula was updated on the same day as well) 

If the consensus is that we should revert to the current instructions, I can do that, but I think we could do much better with simplifying the installation experience.

Reviewer: I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac.  For now, until we decide a larger change, let's use the standard "download CLI" instructions.

Reviewer 2: ```(e.g. the 1.22 was out on May 13th and the brew formula was updated on the same day as well)```

This illustrates my point. istioctl being upgraded ahead of the cluster can introduce some unexpected or bad behavior. Given the upgrade trends we see when Mitch did his surveys, many users are at least N-1, some further. istioctl should not be on a different minor version.

> I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac. For now, until we decide a larger change, let's use the standard "download CLI" instructions.

> I support the goal of making this easier, but I also need to point out that using we can't assume people are using a Mac. For now, until we decide a larger change, let's use the standard "download CLI" instructions.

Homebrew is not just for Macs. I use it on my Linux systems. My point is consistency across docs, and that the version of istioctl should match the version of Istio installed in the clusters, something `brew` cannot determine on its own.

Author: Are we expecting upgrading users to go through the "Getting Started" page? Perhaps we need a section that's a quick start guide or something aimed at new users?

My assumption was that someone going through the "getting started" page probably doesn't have Istio installed at all. But I might be wrong.

Reviewer 2: What about after getting started? IE, if brew upgrades istioctl, but Istio in their env is not upgraded. Manifests are baked into istioctl, and they're specific to that minor version. Setting someone up for a bad time, imo.

Author: Brew won't automatically update it, user has to run update/upgrade, but I see your point. 

I am not necessarily advocating for Brew per-se, but for an easier way to install the CLI. E.g. even if it was an existing downloadScript that downloads the binary only and puts it in ~/.istio folder or something and doesn't require me to set the path and it doesn't "pollute" my current folder with yaml manifests.

Author: Resolving this - we're keeping the existing installation method in the guide. If there are changes that we want to make, they should be made in the installer first.

{{< text bash >}}
$ brew install istioctl
{{< /text >}}

{{< quote >}}
If you don't have Homebrew, follow these instructions to download and install the Istio CLI:

{{< text syntax=bash snip_id=none >}}
$ curl -L https://istio.io/downloadIstio | sh -
$ cd istio-{{< istio_full_version >}}
$ export PATH=$PWD/bin:$PATH
{{< /text >}}

{{< /quote >}}

Verify the CLI is installed correctly by running the following command:

{{< text syntax=bash snip_id=none >}}

> $ istioctl version

Reviewer: https://github.com/istio/istio/issues/51296

no ready Istio pods in "istio-system"
{{< istio_full_version >}}
{{< /text >}}

## 2. Install Istio onto your cluster

Assuming your cluster is prepared, you can install Istio using the ambient mode profile. Run the following command:

{{< text syntax=bash snip_id=install_ambient >}}
$ istioctl install --set profile=ambient --skip-confirmation
{{< /text >}}

Note that it might take a minute for the Istio components to be installed. Once the installation completes you’ll get the following output that indicates all components have been installed successfully!

{{< text syntax=plain snip_id=none >}}
✔ Istio core installed
✔ Istiod installed
✔ CNI installed
✔ Ztunnel installed
✔ Installation complete
{{< /text >}}

You can verify the installed components using the following command:

{{< text syntax=bash snip_id=verify_install >}}
$ istioctl verify-install

> 

Reviewer: we can elide most of the items from this list?

Author: In my notes, I wrote "perhaps we need a --short option for the verify-install command" :) 

But yes, I'll remove this as the initial installation already outputs the checkmarks and verifies everything is installed.
